### ISSUE TITLE
Using machine learning to deal with Phishing and Spam Detection: An overview

### ISSUE THREAD

**Issue by AndrewBreyen** [team3-cs373/resources#63]
### Title
Using machine learning to deal with Phishing and Spam Detection: An overview

### URL
https://doi.org/10.1145/3386723.3387891

### Summary 
As mobile technologies improve, attack technologies also are rapidly improving. 

**ML Phishing Detection:** (1)
The authors of a cited source devised a new rule based method to classify web pages using two new sets of features proposed to detect phishing web pages in internet banking, -- four to assess the identity of the resources of the web page, afour to identify the protocol for accessing resource elements and nine characteristics from related work. 

The ML Phishing model is based on:
* Characteristics extraction by examining the content of the page and the DOM (Document Object Model)
* Classification of web pages using the SVM (Support Vector Machine) algorithm with a three degree polynomial kerneland cross-validation by 10 folds.
* Rules extraction from the model using the SVM and DT(Decision tree) algorithms.
* The implementation of an extension to the Google Chrome browser.

The experiments show that the accuracy of detection increases, and error rate reduces by using the 17 characteristics presented. Positive effect of six characteristics on eight proposed according to sensitivity analysis. Method is more precise when using rules directly from the model than extracted from training data. 

**Phishing detection based on hyperlink information:** (2)
Data set was divided into 12 categories relating to hyperlinks. 
The performance of the proposed solution was evaluated with different algorithms: SMO (Sequential Minimal Optimization), NB(Naive Bayes), RF (Random Forest), SVM, Adaboost, ANN (Artificial Neural Networks), C4.5 and RL (Logistic Regression) using 10-fold cross-validation and the following datasets: Phish tank data set(2018), Stuffgate Free Online Website Analyzer and Alexa top websites (2018), list of online payment service providers (2018).

**Intelligent rule-based phishing website classification:** (3)
17 different features were automatically extracted to detect phishing web pages. Divided into 4 groups. Characteristics based on the address bar, Abnormal characteristics, Characteristics based on HTML/Javascript and Characteristics based on the domain. The characteristics are extracted automatically using Javascript programs and PHP scripts. 

![image](https://user-images.githubusercontent.com/7895581/106372540-3e53df80-6336-11eb-83ec-b6ad944f5410.png)
![image](https://user-images.githubusercontent.com/7895581/106372557-5f1c3500-6336-11eb-96b9-b14d72fd491f.png)

**Spam Detection**
Spam can be defined as an electronic communication sent automatically, unwanted, unsolicited, with unwanted content, received without the full consent of the recipient. Spam can take many forms; Spam in email, spam in SMS, spam in reviews, etc.

The proposed approach is LSTM with a semantic layer added, itis a layer for preprocessing before using the classifier whose text is converted into semantic word vectors using word2vec, WordNet and ConceptNet, after they have introduces the vector as input into the LSTM unit with the output of the previous LSTM unit. This is repeated with each input sentence and in this way the LSTM units continue to record the important characteristics.

To improve spam detection systems, the authors have given four steps: 
1. Step 1: the NB Classifier has classified the training set into two categories: spam and non-spam.
2. Step 2: SVM has built a hyperplane to divide the two categories
3. Step 3: Verification of the results with elimination of anys ample whose result from its nearest neighbor is different.
4. Step 4: the NB Classifier is used for the second time to classify the new training set and detect spam

**CONCLUSION:**
results of the comparative approaches demonstrate the effectiveness of machine learning in network security. In addition, there is still open research and future work on the use of machine learning to improve network security. We have presented work related to phishing and spam only, and there are other types of attacks that we will study in a future work such as machine learning used to detect malware attacks.

### Citation
@inproceedings{10.1145/3386723.3387891,
author = {El Kouari, Oumaima and Benaboud, Hafssa and Lazaar, Saiida},
title = {Using Machine Learning to Deal with Phishing and Spam Detection: An Overview},
year = {2020},
isbn = {9781450376341},
publisher = {Association for Computing Machinery},
address = {New York, NY, USA},
url = {https://doi.org/10.1145/3386723.3387891},
doi = {10.1145/3386723.3387891},
abstract = {Cybersecurity is a growing field that requires a lot of attention due to the remarkable progress made in social networks, cloud and web technologies, online banking, mobile environment, smart networks, etc. Various approaches have been developed to solve many computer security problems, including those based on machine learning. This paper examines and highlights the various works using machine learning in network security. Two types of detection are discussed. Phishing detection and Spam detection. For each type, related work is presented and some proposed methods in the literature are compared taking into account their accuracy and other characteristics.},
booktitle = {Proceedings of the 3rd International Conference on Networking, Information Systems &amp; Security},
articleno = {72},
numpages = {7},
keywords = {Machine Learning, Cyber Security, Network Security, Phishing detection, Spam detection},
location = {Marrakech, Morocco},
series = {NISS2020}
}
